dace/subsets.py

```diff
@@ -62,6 +63,11 @@ def coord_at(self, i):
         """
         raise NotImplementedError
 
+    @property
+    def free_symbols(self) -> Set[str]:
+        """ Returns a set of undefined symbols in this subset. """
+        raise NotImplementedError
```

Comment:
Minor: you can make this a class method and print the class which hasn't implemented it

Reply:
While I cannot make this a class method (as it's a property), the latest commit prints out the class name that does not implement this method.